Lib/test/libregrtest/main.py

```diff
@@ -432,6 +433,26 @@ def display_header(self):
         print("== encodings: locale=%s, FS=%s"
               % (locale.getpreferredencoding(False),
                  sys.getfilesystemencoding()))
+        rl_version_cmd = """if 1:
+        try:
+            import readline
+        except ImportError:
+            readline = None
+        if readline is not None:
+            print(f"{readline._READLINE_VERSION:#06x}")
+            is_editline = readline.__doc__ and "libedit" in readline.__doc__
+            print("editline" if is_editline else "GNU readline")
+        """
+        rl_output = subprocess.run([sys.executable, "-c", textwrap.dedent(rl_version_cmd)],
+                                   stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        if rl_output.returncode == 0:
+            try:
+                rl_version, rl_implementation = rl_output.stdout.decode().splitlines()
+            except ValueError:
+                pass
+            else:
+                print(f"== readline version: {rl_version}")
+                print(f"== readline implementation: {rl_implementation}")
```

Comment:
I like it. Maybe just add version somewhere? "== readline: version 0x0602 (GNU readline)"